libs/contrib/Text/Lexer.idr

```diff
@@ -48,6 +48,40 @@ export
 isNot : Char -> Lexer
 isNot x = Pred (/=x)
 
+||| Recognise a character case-insensitively
+like : Char -> Lexer
+like x = Pred (\y => toUpper x == toUpper y)
+
+export
+is_ci : Char -> Lexer
+is_ci = like
+%deprecate Text.Lexer.is_ci "Please use `like`."
```

Comment:
The function `is_ci` and the thing it deprecates (`like`)are new additions. Would it not be worth just removing `is_ci`?

Reply:
I see why this might be confusing. I was trying to accommodate new users of the Lexer library, while still preferring names that maximize readability.

I prefer the names `like` and `approx`. They flow really well when reading lexer code. But they will be difficult to find for someone asking "is there a case insensitive version of `exact`?"

If you think the documentation comments on `like` and `approx` are enough to make them easy to find (via `:apropos`), then these can be removed. It will just be harder to find the case-insensitive functions via `browse`.

Reply:
TBH, i found `is_ci` to be confusing enough not to think `ci` implied case insensitive. I think function 'discovery' will be a issue regardless of name unless the name is precisely: `is_but_case_insensitive` or you place it under a different namespace e.g. `CaseInsensitive.is`. Both these options are not ideal. 

I think the names `like` and `approx` with sufficient documentation will be fine.

Reply:
Fair enough. I'll take them out.